proto/substrait/type.proto

```diff
@@ -99,8 +99,9 @@ message Type {
   }
 
   message Timestamp {
-    uint32 type_variation_reference = 1;
-    Nullability nullability = 2;
+    optional int32 precision = 1;
```

Comment:
The numbers here should not be reordered as that affects backwards compatibility.